sregistry/client/__init__.py

```diff
@@ -193,6 +193,10 @@ def get_parser():
                        help='name of image, in format "library/image"', 
                        type=str, required=True)
 
+    push.add_argument("--public-read", dest='public_read',
+                       help='make image publicly readable, if supported by backend',
+                       default=False, action='store_true')
+
```

Comment:
So - since this is the push command for any general client (Google, Amazon, Dropbox, etc.) we can't add custom arguments for just one client. What you've done below in all the files is to add the argument to give, but it's misleading because it's not actually supported.

I'm not sure how to best resolve this - in the past I've used the "extra" arguments to add customizations for one client. It's sort of the case that this (could be) a variable for other clients too, but then what is the criteria for adding a variable (for just one client)? 

Another option (that I've done with others) is to have this be an environment setting that is parsed by the client.
Another option would be to entirely separate the clients to be specific to each, but that seems redundant (and it removes the functionality that it can handle any one, and then "discover" the active based on an envar.

So - this doesn't seem right, but I don't have other suggestions that I think you would prefer. Thoughts?

Reply:
Hi @vsoch!

I have indeed added this option hoping that one day it may be used by the other backends as well. If you think I have been too optimistic, I don't mind rewriting it and use an appropriate environment variable instead: something like `SREGISTRY_S3_OBJECT_ACL` or so, which would let the user pick exactly which setting he/she prefers. What do you think?

Reply:
I would too want the option to be added across clients, but the worry now is that the user *thinks* the flag is doing something (but it's not). So I think for now, that's probably the best option. And add the variable to the S3 docs so the user knows about it :)

Reply:
Ok, think you will like this one more.